homeassistant/components/homekit_controller/button.py

```diff
@@ -115,3 +123,45 @@ async def async_press(self) -> None:
         key = self.entity_description.key
         val = self.entity_description.write_value
         return await self.async_put_characteristics({key: val})
+
+
+class HomeKitEcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):
+    """Representation of a Button control for Ecobee clear hold request."""
+
+    _attr_name = "Clear Hold"
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return []
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        prefix = ""
+        if name := super().name:
+            prefix = name
+        return f"{prefix} {self._attr_name}"
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        key = self._char.type
+
+        for val in (False, True):
+            await self.async_put_characteristics({key: val})
+
+
+BUTTON_ENTITY_CLASSES: dict[str, type] = {
+    CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD: HomeKitEcobeeClearHoldButton,
+}
+
+
+# For legacy reasons, "built-in" characteristic types are in their short form
+# And vendor types don't have a short form
+# This means long and short forms get mixed up in this dict, and comparisons
+# don't work!
+# We call get_uuid on *every* type to normalise them to the long form
+# Eventually aiohomekit will use the long form exclusively amd this can be removed.
+for k, v in list(BUTTON_ENTITY_CLASSES.items()):
```

Comment:
Don't bother with this cludge. Its not needed here because clear hold is already a normalised uuid. And i'm working to get rid of all the other occurrences of this this next release cycle so there is no point in adding it.

Reply:
Removed